src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -122,6 +122,12 @@ end = struct
     add comps t
 end
 
+type inotify =
```

Comment:
Could you add some comments here, explaining what `awaiting_creation` and `mutex` fields are for?